accern_xyme/util.py

```diff
@@ -220,24 +226,130 @@ def async_compute(
+                add_more = max(
+                    max_buff - len(ids),
+                    max_buff - max_count)
+                cur = arr[pos:pos + add_more]
+                pos += len(cur)
+                for block_ix in range(0, len(cur), block_size):
+                    ids.update({
+                        cur_id: cur_ix + start_pos + block_ix
+                        for (cur_ix, cur_id) in enumerate(
+                            start(cur[block_ix:block_ix + block_size]))
+                    })
+                with cond:
+                    cond.notify_all()
+        finally:
+            end_produce[0] = True
+            with cond:
+                cond.notify_all()
+
+    def consume() -> None:
+        while not done[0]:
+            with cond:
+                cond.wait_for(
+                    lambda: exc[0] is not None or done[0] or len(ids) > 0)
+            do_wait = False
+            while ids:
+                if do_wait:
+                    time.sleep(1)
+                do_wait = True
+                status = get_status([
+                    hnd for hnd, _ in ids.items()
+                ])
+                for (t_id, t_status) in status.items():
+                    # print("t_id, t_status", t_id, t_status, t_id in ids)
+                    if t_status in ("waiting", "running"):
+                        continue
+                    do_wait = False
+                    try:
+                        t_ix = ids.pop(t_id)
+                        res[t_ix] = get(t_id)
+                    except NotFoundError:
+                        pass
+                    except ServerSideError as e:
+                        if exc[0] is None:
+                            exc[0] = e
+                    except KeyError as e:
+                        pass
+            with cond:
+                cond.notify_all()
+
+    prod_th = threading.Thread(target=produce)
+    prod_th.start()
+    consume_th = threading.Thread(target=consume)
+    consume_th.start()
+    yield_ix = 0
+    while yield_ix < len(arr):
+        with cond:
+            cond.wait_for(lambda: exc[0] is not None or bool(res))
+        if exc[0] is not None:
+            break
+        try:
+            while res:
+                yield res.pop(yield_ix)
+                yield_ix += 1
+        except KeyError:
+            pass
+    if exc[0] is not None:
+        with cond:
+            cond.wait_for(lambda: end_produce[0])
+    done[0] = True
+    with cond:
+        cond.notify_all()
+    prod_th.join()
+    consume_th.join()
+    raise_e = exc[0]
+    if isinstance(raise_e, BaseException):
+        raise raise_e  # pylint: disable=raising-bad-type
+
+
+class ServerSideError(Exception):
+    def __init__(self, message: str) -> None:
+        self._message = message
+        super().__init__(self._message)
+
+    def __str__(self) -> str:
+        return f"Error from xyme backend: \n{self._message}"
+
+
+class NotFoundError(Exception):
```

Comment:
so the plan was to actually raise an exception on a 404 (in any request method) and then convert that exception to a keyerror for the dynamic getter